power_grid_model_c/power_grid_model/include/power_grid_model/job_dispatch.hpp

```diff
@@ -142,9 +146,9 @@ class JobDispatch {
               typename RecoverFromBadFn>
         requires std::invocable<std::remove_cvref_t<RunFn>, Args const&...> &&
                  std::invocable<std::remove_cvref_t<SetupFn>, Args const&...> &&
-                 std::invocable<std::remove_cvref_t<WinddownFn>, Args const&...> &&
+                 std::invocable<std::remove_cvref_t<WinddownFn>> &&
                  std::invocable<std::remove_cvref_t<HandleExceptionFn>, Args const&...> &&
-                 std::invocable<std::remove_cvref_t<RecoverFromBadFn>, Args const&...>
+                 std::invocable<std::remove_cvref_t<RecoverFromBadFn>>
```

Comment:
the original reason i kept the scenario index (and other things variadically`...`) in here (and elsewhere) was because it generalizes to any arbitrary job. For instance, a verbose logger wrapper implementation may implement logging both `Setting up scenario 123` and `Winding down scenario 123`. Asymmetry between setup and winddown removes that support

Reply:
We also had another idea of having it called using `non_existant_scenario=-1`. That feels okay.